packages/cache/src/internal/tar.ts

```diff
@@ -11,17 +11,24 @@ async function getTarPath(
   args: string[],
   compressionMethod: CompressionMethod
 ): Promise<string> {
+  let tarPath = await io.which('tar', true)
   switch (process.platform) {
     case 'win32': {
+      const gnuTar = `${process.env['PROGRAMFILES']}\\Git\\usr\\bin\\tar.exe`
       const systemTar = `${process.env['windir']}\\System32\\tar.exe`
-      if (compressionMethod !== CompressionMethod.Gzip) {
+      if (existsSync(gnuTar)) {
+        // Use GNUtar as default on windows
+        args.push('--force-local')
+        tarPath = gnuTar
```

Comment:
Here `${process.env['PROGRAMFILES']}\\Git\\usr\\bin\\tar.exe` is used as hard-coded path but isGnuTarInstalled() also looks at direct tar command as well. I feel this is an inconsistency. can we use a common logic to find gnutar path and reuse it everywhere?

Reply:
My suggestion would be to have a common function named `getGnuTarPath` and return path else null. Use this function both here as well as in isGnuTarInstalled

Reply:
Okay. That makes sense